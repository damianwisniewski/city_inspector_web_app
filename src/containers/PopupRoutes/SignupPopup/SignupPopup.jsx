import React, { Component } from 'react'
// import PropTypes from 'prop-types'

import { Form, Input, Group, Checkbox } from '../../../components/FormComponents'
import Button from '../../../components/CommonComponents/Button/Button'

import './SignupPopup.scss'

// import Status from '../../../components/CommonComponents/Status/Status'

class SignupPopup extends Component {
	render() {
		const { closePopup } = this.props

		return (
			<div id='SignupPopup' className='signup'>
				<h2 className='signup__header'>Rejestracja</h2>
				{/* <Status id='signup-status' type='error' message='Niepoprawny email lub hasło' /> */}
				<p className='signup__description'>Wpisz następujące dane aby się zarejestrować</p>
				<Form className='signup__form'>
					{/* Required data */}
					<div className='signup__required'>
						<Input
							id='email-field'
							label='E-mail*'
							placeholder='Podaj swój adres email...'
							floatingLabel
							type='email'
							autoComplete='username'
							pattern='.+@.+'
							required
						/>
						<Input
							id='password-field'
							label='Hasło*'
							placeholder='Podaj hasło...'
							floatingLabel
							type='password'
							name='password'
							autoComplete='new-password'
							required
						/>
						<Input
							id='repeat-password-field'
							label='Powtórz hasło*'
							placeholder='Powtórz hasło...'
							floatingLabel
							type='password'
							autoComplete='current-password'
							required
						/>
						<Input
							id='nickname-field'
							label='Nickname*'
							placeholder='Podaj swój adres email...'
							floatingLabel
							type='text'
							autoComplete='username'
							pattern='.+@.+'
							required
						/>
					</div>
					{/* Optionals data */}
					<Group name='opcjonalnie'>
						<Input
							id='name-field'
							label='Imię'
							placeholder='Podaj swoje imię...'
							floatingLabel
							type='text'
							autoComplete='given-name'
						/>
						<Input
							id='surname-field'
							label='Nazwisko'
							placeholder='Podaj swoje nazwisko...'
							floatingLabel
							type='text'
							autoComplete='family-name'
						/>
						<Input
							id='city-field'
							label='Miasto'
							placeholder='Podaj swoje miasto...'
							floatingLabel
							type='text'
							name='city'
							autoComplete='address-level2'
						/>
					</Group>
					{/* Agreements checkbox */}
					<div className='signup__agreements-wrapper'>
						<Checkbox
							id='agreement_1'
							label='Czy wyrażasz zgodę na otrzymywanie informacji mailowych od serwisu "City Inspector", dotyczących zmian w serwisie oraz aktualizacji zgłoszeń.'
						/>
						<Checkbox
							id='agreement_2'
							label='Czy twoje dane takie jak e-mail, imię i nazwisko oraz miasto, mają być widocznie dla innych zarejestowanych użytkowników.'
						/>
					</div>
					{/* Buttons*/}
					<div className='signup__buttons-wrapper'>
						<Button id='inform-close-button' color='white' onClick={closePopup}>
							Anuluj
						</Button>
						<Button id='submit-button' type='submit' color='blue'>
							Zaloguj
						</Button>
					</div>
				</Form>
			</div>
		)
	}
}

// SignupPopup.propTypes = {}

export default SignupPopup
