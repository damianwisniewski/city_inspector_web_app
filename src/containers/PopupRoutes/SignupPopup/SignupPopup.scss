@import '../../../assets/styles/index';

.signup {
	min-width: 310px;
	max-width: 600px;
	text-align: center;

	&__header {
		font-size: 20px;
		font-family: $secondary_font;
		color: $light_gray;
		background-color: $primary_blue;
		margin: 15px 10px;
		padding: 10px;
		border-radius: 5px;
		box-shadow: $inset-shadow;
		text-shadow: $text-shadow;
	}

	&__description {
		font-size: 15px;
		margin: 10px;
		color: $dark_gray;
	}

	&__form {
		display: flex;
		flex-wrap: wrap;

		& > * {
			flex: 1;
		}
	}

	&__buttons-wrapper {
		margin-top: 20px;
		margin-bottom: 10px;
		min-width: 100%;
	}

	&__agreements-wrapper {
		text-align: center;
		padding: 10px;
		min-width: 100%;
	}

	@media only screen and (max-width: 550px) {
		&__required {
			min-width: 100%;
			margin-bottom: 20px;
		}
	}
}
