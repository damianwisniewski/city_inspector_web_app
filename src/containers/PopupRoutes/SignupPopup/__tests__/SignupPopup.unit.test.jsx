import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'

import SignupPopup from '../SignupPopup'

configure({ adapter: new Adapter() })

describe('<SignupPopup />', () => {
	let SignupPopupElement

	test('should render correctly SignupPopup', () => {
		SignupPopupElement = shallow(<SignupPopup />)
		expect(toJson(SignupPopupElement)).toMatchSnapshot()
	})
})
