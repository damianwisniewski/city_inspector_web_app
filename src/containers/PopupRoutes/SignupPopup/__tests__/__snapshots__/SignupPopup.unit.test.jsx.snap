// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignupPopup /> should render correctly SignupPopup 1`] = `
<div
  className="signup"
  id="SignupPopup"
>
  <h2
    className="signup__header"
  >
    Rejestracja
  </h2>
  <p
    className="signup__description"
  >
    Wpisz następujące dane aby się zarejestrować
  </p>
  <Form
    className="signup__form"
  >
    <div
      className="signup__required"
    >
      <Input
        autoComplete="username"
        floatingLabel={true}
        id="email-field"
        label="E-mail*"
        pattern=".+@.+"
        placeholder="Podaj swój adres email..."
        required={true}
        type="email"
      />
      <Input
        autoComplete="new-password"
        floatingLabel={true}
        id="password-field"
        label="Hasło*"
        name="password"
        placeholder="Podaj hasło..."
        required={true}
        type="password"
      />
      <Input
        autoComplete="current-password"
        floatingLabel={true}
        id="repeat-password-field"
        label="Powtórz hasło*"
        placeholder="Powtórz hasło..."
        required={true}
        type="password"
      />
      <Input
        autoComplete="username"
        floatingLabel={true}
        id="nickname-field"
        label="Nickname*"
        pattern=".+@.+"
        placeholder="Podaj swój adres email..."
        required={true}
        type="text"
      />
    </div>
    <Group
      direction="column"
      name="opcjonalnie"
    >
      <Input
        autoComplete="given-name"
        floatingLabel={true}
        id="name-field"
        label="Imię"
        placeholder="Podaj swoje imię..."
        type="text"
      />
      <Input
        autoComplete="family-name"
        floatingLabel={true}
        id="surname-field"
        label="Nazwisko"
        placeholder="Podaj swoje nazwisko..."
        type="text"
      />
      <Input
        autoComplete="address-level2"
        floatingLabel={true}
        id="city-field"
        label="Miasto"
        name="city"
        placeholder="Podaj swoje miasto..."
        type="text"
      />
    </Group>
    <div
      className="signup__agreements-wrapper"
    >
      <Checkbox
        id="agreement_1"
        label="Czy wyrażasz zgodę na otrzymywanie informacji mailowych od serwisu \\"City Inspector\\", dotyczących zmian w serwisie oraz aktualizacji zgłoszeń."
      />
      <Checkbox
        id="agreement_2"
        label="Czy twoje dane takie jak e-mail, imię i nazwisko oraz miasto, mają być widocznie dla innych zarejestowanych użytkowników."
      />
    </div>
    <div
      className="signup__buttons-wrapper"
    >
      <Button
        className=""
        color="white"
        id="inform-close-button"
        onClick={[Function]}
        type="button"
      >
        Anuluj
      </Button>
      <Button
        className=""
        color="blue"
        id="submit-button"
        onClick={[Function]}
        type="submit"
      >
        Zaloguj
      </Button>
    </div>
  </Form>
</div>
`;
